scripts/cmake/z_vcpkg_fixup_install_name_rpath.cmake

```
@@ -0,0 +1,419 @@
+  if(get_dep_install_names_rv EQUAL 0)
+    # Strip the first line
+    string(REGEX REPLACE "^.*:\n" "" get_dep_install_names_ov "${get_dep_install_names_ov}")
+    # Extract the install names of the dependencies
+    string(REGEX MATCHALL "[^\n\t]+(.dylib|.so)" dependencies "${get_dep_install_names_ov}")
+
+    z_vcpkg_get_mach_o(type FILE "${arg_FILE}")
+    if(type EQUAL 1)
+      list(LENGTH dependencies n)
+      math(EXPR n "${n}-1")
+      list(SUBLIST dependencies 1 ${n} dependencies)
+    endif()
+
+    set("${out_var}" "${dependencies}" PARENT_SCOPE)
+  else()
+    message(WARNING "otool -L ${arg_FILE}, failed: \n${get_dep_install_names_ov}")
+  endif()
+endfunction()
+
+# Define a function to fix the install names and rpaths in Mach-O files in package dir
+# A common package dir stucture:
+# /opt/vcpkg/packages/icu_arm64-osx-dynamic/
+# ├── debug
+# │  └── lib
+# │       ├── libicudata.72.1.dylib
+# │       ├── libicudata.72.dylib -> libicudata.72.1.dylib
+# │       ├── libicudata.dylib -> libicudata.72.1.dylib
+# │       ├── ...
+# │       └── pkgconfig
+# ├── include
+# │   └── unicode
+# │       ├── alphaindex.h
+# │       ├── ...
+# │       └── vtzone.h
+# ├── lib
+# │   ├── libicudata.72.1.dylib
+# │   ├── libicudata.72.dylib -> libicudata.72.1.dylib
+# │   ├── libicudata.dylib -> libicudata.72.1.dylib
+# │   ├── ...
+# │   └── pkgconfig
+# │       ├── ...
+# │       └── icu-uc.pc
+# ├── share
+# │   └── icu
+# └── tools
+#     └── icu
+#         └── bin
+#             └── derb 
+# Here, it'wll do fix:
+# 1. for shared libraries in `lib` and `debug/lib`
+#    - fix its own install names to `@rpath` prefix 
+#    - fix its dependent shared library install name to `@rpath` prefix
+#       - include libraries only from the same port, assuming other libraries from other managed ports by vcpkg are already fixed!
+#       - exclude system libraries in `/usr/lib/` and `/System/Library`!
+#    - fix rpath to `@loader`(or using absolute path `CURRENT_INSTALLED_DIR`, like `/opt/vcpkg/installed/arm64-osx-dynamic/lib` or `/opt/vcpkg/installed/arm64-osx-dynamic/debug/lib`)
+# 2. for exe in `tools/{package}/bin/`
+#    - fix its dependent shared library install name to `@rpath` prefix
+#       - include libraries only from the same port in lib/, assuming other libraries from other managed ports by vcpkg are already fixed!
+#       - exclude system libraries in `/usr/lib/` and `/System/Library`!
+#    - fix rpath to `@loader/../../../`  to be relative with `lib/`
+function(z_vcpkg_fixup_install_name_rpath_in_dir)
+  find_program(
+    install_name_tool_cmd
+    NAMES install_name_tool
+    DOC "Absolute path of install_name_tool cmd"
+  )
+
+  if(NOT install_name_tool_cmd)
+    message(WARNING " install_name_tool not found!")
+    return()
+  endif()
+
+  find_program(
+    otool_cmd
+    NAMES otool
+    DOC "Absolute path of otool cmd"
+  )
+
+  if(NOT otool_cmd)
+    message(WARNING "otool not found!")
+    return()
+  endif()

```

```suggestion
    DOC "Absolute path of otool cmd"
    REQUIRED
  )
```

This can't depend on what happens to be on the box, it needs to be done, a warning isn't sufficient.

Otherwise we end up binary caching broken bits from a broken machine and restoring them on a hypothetically working machine.